server/src/main/java/org/elasticsearch/http/DefaultRestChannel.java

```diff
@@ -82,66 +80,87 @@ protected BytesStream newBytesOutput() {
 
     @Override
     public void sendResponse(RestResponse restResponse) {
-        // We're sending a response so we know we won't be needing the request content again and release it
-        httpRequest.release();
-
-        final ArrayList<Releasable> toClose = new ArrayList<>(3);
-        if (HttpUtils.shouldCloseConnection(httpRequest)) {
-            toClose.add(() -> CloseableChannel.closeChannel(httpChannel));
-        }
+        try (var ignored = tracer.withScope(this)) {
```

Comment:
What is this actually going to be used for? Could you add some javadoc to `withScope` maybe just so reviewers can understand what the lifecyle needs of this thing are?
If we can help it I would definitely want to avoid adding this kind of complexity to the already complex resource management in this method.

Reply:
I took this wrapping out. I added Javadoc to the `withScope()` method that explains when you might want to use it, but it's not import here.